art/estimators/classification/tensorflow.py

```diff
@@ -800,7 +815,7 @@ def predict(self, x: np.ndarray, batch_size: int = 128, **kwargs) -> np.ndarray:
             )
 
             # Run prediction
-            results[begin:end] = self._model(x_preprocessed[begin:end])
```

Comment:
This and the following similar changes look like a bug to me, as it's now referencing `self.model` instead of `self._model`. Is this intentional?

Reply:
I noticed that we were not using the new model property everywhere and made the updates, but now I think it's better to make these changes in another pull request and I have reverted them here.